pkg/user/repository/user.go

```diff
@@ -0,0 +1,62 @@
+package repository
+
+import (
+	"context"
+	"fmt"
+	"online_shop_backend/pkg/models"
+	"online_shop_backend/pkg/storage"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5/pgtype"
+)
+
+type userRepo struct {
+	storage *storage.Storage
+}
+
+func New(store *storage.Storage) User {
+	return &userRepo{
+		storage: store,
+	}
+}
+
+func (ur *userRepo) Get(ctx context.Context, email string) (models.User, error) {
+	select {
+	case <-ctx.Done():
+		return models.User{}, fmt.Errorf("done with context")
+	default:
+		pool := ur.storage.Pool
+		row := pool.QueryRow(ctx, `SELECT id, name, password FROM users WHERE email=$1`, email)
+		res := models.User{}
+		var password pgtype.Text
+		err := row.Scan(&res.ID, &res.Name, &password)
+		if password.Valid {
+			res.Password = password.String
+		}
+		if err != nil {
+			return models.User{}, fmt.Errorf("ca't get data from db: %w", err)
+		}
+		return res, nil
+	}
+
+}
+
+func (ur *userRepo) Create(ctx context.Context, user models.User) (uuid.UUID, error) {
```

Comment:
Can we instead return the whole models.User structure?

Reply:
Sure thing.